Src/Common/CorrelationIdLookupHelper.cs

```diff
@@ -119,18 +133,45 @@ public bool TryGetXComponentCorrelationId(string instrumentationKey, out string
                             {
                                 this.GenerateCorrelationIdAndAddToDictionary(instrumentationKey, appId.Result);
                             }
+                            catch (AggregateException ex)
+                            {
+                                ex.Flatten();
+                                if (ex.InnerExceptions != null && ex.InnerExceptions.Count > 0 && ex.InnerExceptions[0] != null)
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex.InnerExceptions[0]);
+                                }
+                                else
+                                {
+                                    this.RegisterFailure(instrumentationKey, ex);
+                                }
+                            }
                             catch (Exception ex)
                             {
-                                CrossComponentCorrelationEventSource.Log.FetchAppIdFailed(this.GetExceptionDetailString(ex));
+                                this.RegisterFailure(instrumentationKey, ex);
```

Comment:
In that case just move the logic to unwrap aggregate exception in RegisterFailure call. Almost everything stays the same as it was before develop except for the following: 
1. GetAwaiter is gone - use .Result directly. 
2. Register Failure can check if the exception is Aggreagate and unwrap it before logging it.